core/trino-main/src/main/java/io/trino/operator/function/TableFunctionOperator.java

```diff
@@ -285,21 +294,36 @@ public TableFunctionOperator(
         PagesIndex pagesIndex = pagesIndexFactory.newPagesIndex(sourceTypes, expectedPositions);
         HashStrategies hashStrategies = new HashStrategies(pagesIndex, partitionChannels, prePartitionedChannels, sortChannels, sortOrders, preSortedPrefix);
 
-        this.outputPages = pageBuffer.pages()
-                .transform(new PartitionAndSort(pagesIndex, hashStrategies, processEmptyInput))
-                .flatMap(groupPagesIndex -> pagesIndexToTableFunctionPartitions(
-                        groupPagesIndex,
-                        hashStrategies,
-                        tableFunctionProvider,
-                        session,
-                        functionHandle,
-                        properChannelsCount,
-                        passThroughSourcesCount,
-                        requiredChannels,
-                        markerChannels,
-                        passThroughSpecifications,
-                        processEmptyInput))
-                .flatMap(TableFunctionPartition::toOutputPages);
+        if (preferStreaming && passThroughSpecifications.isEmpty() && sortOrders.isEmpty()) {
```

Comment:
For streaming execution, we must partition the data (it might be pre-partitioned but there's no guarantee it's fully partitioned), and create a new data processor for each partition.